### ISSUE TITLE
Flexible Content within Flexible Content

### ISSUE THREAD

**Issue by peter-brennan** [MarieComet/mc-acf-flexible-template#8]
Hey I have been using your plugin for a while now and it is awesome, thank you for your hard work!

I have changed a few things with how my content is captured in the backend and I have run into an issue.

The main part of my content is a flexible content field where I can add sections. One of the sections also contains a flexible content field with in it that I add more sub sections.

When I save the main section flexible content items as a template, the inner flexible content fields become messed up and breaks the content.

<img width="903" alt="screen shot 2018-06-25 at 7 19 12 pm" src="https://user-images.githubusercontent.com/25376577/41841864-b4ee2a96-78ac-11e8-92f8-1976b1c6edac.png">

The image above shows the saved flexible content template. As you can see the 'Page Content' field is a flexible content field has the 'Columns' Flexible Content field, with two items assigned. The layout and content is saving to the ACF Template CPT correctly.

<img width="1354" alt="screen shot 2018-06-25 at 7 16 58 pm" src="https://user-images.githubusercontent.com/25376577/41841994-08c0c73c-78ad-11e8-9932-41aaf8463ca8.png">

The image above shows what happens when I import the saved template back into the Page Content Flexible Content area. As you can see the inner Flexible Content field that contains the two Wysiwyg Editor fields has duplicated. It appears the whole inner Flexible Content field has duplicated as you can see two "Add Column" buttons in it.

I hope you understand my issue. I love using your plugin. If you know how we can fix this issue and help is much appreciated.

Thanks in advanced, Peter

**Comment by MarieComet** [comment]
Hi Peter, can you share and export (json or php) of your groups fields ?
It will be easier for me to reproduce this issue.
Thanks you.

**Comment by peter-brennan** [comment]
Hi MarieComet thank you for your fast reply.

[acf-export-2018-06-25 (1).json.zip](https://github.com/MarieComet/mc-acf-flexible-template/files/2132735/acf-export-2018-06-25.1.json.zip)

Above is the export of a few of the groups I am using to create my page fields. This might not work for you out of the box as I am using a purchased plugin call Component fields, here is a link to that. https://acf-component-field.gummi.io/

All that is is basically a repeater field that allows you to create reusable components. If you need me to email you a copy of the plugin for testing purposes please let me know.

If there is anything else I can send to help you debug also let me know.

Cheers

**Comment by MarieComet** [comment]
Yes, I need the Component field plugin too, here is my email : hello@mariecomet.fr

**Comment by peter-brennan** [comment]
Hi Marie,

 

Thank you very much for looking into this so fast. I am very grateful.

 

I have attached the ACF Component Plugin in this email.

 

Kind Regards,

 

Peter Brennan

**Comment by MarieComet** [comment]
I didn't see any attached files, maybe GitHub do not allow them in answer emails.

**Comment by peter-brennan** [comment]
Oh ok, I just resent it in a new email, let me know how you go.

**Comment by peter-brennan** [comment]
Hi Marie,

I have recreated the fields so it is not using the component plugin and it appears to be the same issue..

I have attached the json file below, it might be easier to debug. 

[acf-export-2018-06-25 (2).json.zip](https://github.com/MarieComet/mc-acf-flexible-template/files/2132856/acf-export-2018-06-25.2.json.zip)

As you can see in the image below, it appears that issue is it is not initiating the inner flexible content field.

<img width="1089" alt="screen shot 2018-06-25 at 9 13 06 pm" src="https://user-images.githubusercontent.com/25376577/41847287-d3fe770a-78bc-11e8-8821-547d255f5062.png">

**Comment by MarieComet** [comment]
Ok, I can reproduce with and without the Component field plugin.
There is a problem with nested flexibles/repeater fields, I've tried both situations : 

- Parent (flexible)
-- Child (flexible)
> Not ok.

- Parent (repeater)
-- Child (flexible)
> Not ok.

I didn't see this bug before, maybe some changes was introduced in ACF core.
I will do some tests in the coming days and come back to you.

**Comment by MarieComet** [comment]
Hi @peter-brennan 
After some research, I can't find how to resolve this bug.
All I can sugget to you is to use other fields structure, without nested flexibles fields, to continue using this plugin...
Sorry about that !

**Comment by peter-brennan** [comment]
Hi Marie,
No worries at all, I have had a look at the plugin and I have a fair understanding of its workings so I will have a look into the issue when I get some spare time. If we could leave this open in case anyone runs into the issue before it is resolved. Thank you for taking the time to address it.

**Comment by Stuckya** [comment]
Hello @MarieComet 
I am also having this issue and would love to continue using your plugin with my nested flexible content. Let me know how I can help!

**Comment by sethwhitaker** [comment]
We also use nested Flex content fields for sections like the OP had and its doing the same thing. Which makes this awesome plugin unusable for us. We really love it though and would love to see it fixed

**Comment by mandfolk** [comment]
Hi all

Is there any update on this issue?

I could really use this feature as well on my projects

Thanks

Michael
